### PR TITLE
Add deployment/undeployment of RBAC, non-k8s API, clusters UI

### PR DESCRIPTION
**Pull request by vadimeisenbergibm**

### CODE REVIEW

File: deploy/deploy_hub_of_hubs.sh [hunk 1]
```diff
@@ -52,3 +52,34 @@ curl -s "https://raw.githubusercontent.com/open-cluster-management/hub-of-hubs-s
...
+    REGISTRY=vadimeisenbergibm IMAGE_TAG=$TAG COMPONENT=hub-of-hubs-nonk8s-api envsubst | kubectl apply -f - -n "$acm_namespace"
+
+curl -s "https://raw.githubusercontent.com/open-cluster-management/hub-of-hubs-nonk8s-api/$TAG/deploy/ingress.yaml.template" |
+    COMPONENT=hub-of-hubs-nonk8s-api envsubst | kubectl apply -f - -n "$acm_namespace"
+
+# deploy hub-of-hubs-console using its Helm chart. We could have used a helm chart repository, see https://harness.io/blog/helm-chart-repo,
+# but here we do it in a simple way, just by cloning the chart repo
+
+rm -rf hub-of-hubs-console-chart
+git clone https://github.com/open-cluster-management/hub-of-hubs-console-chart.git
```
Comment (nirrozenbaum) [inline deploy/deploy_hub_of_hubs.sh:78]: git clone may break the script, e.g. if someone runs the script on a machine with no git installed.
we should use instead curl to get resources from git repo.
for example, we can use the following in line 82:
```
curl https://raw.githubusercontent.com/open-cluster-management/hub-of-hubs-console-chart/main/stable/console-chart/values.yaml | sed "s/console: \"\"/console: vadimeisenbergibm\/console:$TAG/g" | helm upgrade ...
```

**Review by nirrozenbaum — COMMENTED** [review]
@vadimeisenbergibm added few comments

**Review by nirrozenbaum — COMMENTED** [review]
another comment - 
would be good to add to README links to RBAC + non k8s api repos..
if our automation scripts are installing it, it would be good if we show it in the list.

File: deploy/deploy_hub_of_hubs.sh [hunk 2]
```diff
@@ -52,3 +52,34 @@ curl -s "https://raw.githubusercontent.com/open-cluster-management/hub-of-hubs-s
...
     SYNC_SERVICE_HOST="$css_sync_service_host" SYNC_SERVICE_PORT="$css_sync_service_port" IMAGE="nirrozenbaumibm/hub-of-hubs-status-transport-bridge:$TAG" envsubst | kubectl apply -f - -n "$acm_namespace"
+
+curl -s "https://raw.githubusercontent.com/open-cluster-management/hub-of-hubs-rbac/$TAG/testdata/data.json" > data.json
+curl -s "https://raw.githubusercontent.com/open-cluster-management/hub-of-hubs-rbac/$TAG/role_bindings.yaml" > role_bindings.yaml
+curl -s "https://raw.githubusercontent.com/open-cluster-management/hub-of-hubs-rbac/$TAG/opa_authorization.rego" > opa_authorization.rego
+
+kubectl delete secret opa-data -n "$acm_namespace" --ignore-not-found
+kubectl create secret generic opa-data -n "$acm_namespace" --from-file=data.json --from-file=role_bindings.yaml --from-file=opa_authorization.rego
+
+rm -rf data.json role_bindings.yaml opa_authorization.rego
```
[the remark this answers is not in the document]
Comment (vadimeisenbergibm) [reply deploy/deploy_hub_of_hubs.sh:63]: testdata contains the roles and role bindings. if you remove lines 56-63 from the script, the REST API and UI will not show any clusters.

these lines simulate the work that some admin could perform. I think for our POC it is nice to automate that work by admin as well, so we will have the same roles that we could demonstrate on all the environments.

File: deploy/deploy_hub_of_hubs.sh (hunk 1 above)
Comment (vadimeisenbergibm) [reply deploy/deploy_hub_of_hubs.sh:78]: I perform the git clone to run the Helm commands on the chart (helm install/upgrade) - I documented it on lines 74-75.

**Comment by vadimeisenbergibm** [comment]
@nirrozenbaum sure, I will add the links directly to the README. Just wanted to merge this PR first.

File: deploy/deploy_hub_of_hubs.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (vadimeisenbergibm) [reply deploy/deploy_hub_of_hubs.sh:63]: actually, let me clarify this issue a bit more, there are two roles here:

1. RBAC implementer - will write `testdata/data.json` with the roles, and `opa_authorization.rego` for authorization of the RBAC components itself.
2. admin will write `role_bindings.yaml` and allocate users to roles, as I did in the demo.

[the remark this answers is not in the document]
Comment (vadimeisenbergibm) [reply deploy/deploy_hub_of_hubs.sh:63]: and to clarify the issue even more, I think in a production system, a Helm chart would be required for the RBAC component,  to create both deployment and the secret from  files. https://helm.sh/docs/chart_template_guide/accessing_files/

[the remark this answers is not in the document]
Comment (nirrozenbaum) [reply deploy/deploy_hub_of_hubs.sh:63]: maybe you can extract RBAC roles setting to a different script?
with this current version, deploy hub of hubs script now includes test data that is not part of hub of hubs but part of a specific demo.
then you could call it something like deploy-rbac-sample.sh or something like that to make sure it's clear that is not part of hub of hubs, but just a sample of a specific rbac setup.

[the remark this answers is not in the document]
Comment (vadimeisenbergibm) [reply deploy/deploy_hub_of_hubs.sh]: @nirrozenbaum I removed test data and added a file with default roles instead. There is a requirement for at least a single role `admin` for the UI to work. The default role `admin`, the default clusterSet `*`, the default role binding (UI service account is admin), the default OPA authorization are part of the general Hub-of-Hubs RBAC implementation, not related to the RBAC demo. I have added to the README a section on configuring RBAC (a link to the RBAC repo).

**Comment by vadimeisenbergibm** [comment]
@nirrozenbaum I have applied your comment regarding testadata and added the repos to the list. Also updated the architecture diagram.

**Review by nirrozenbaum — APPROVED** [review]

**Pull request merged**